lib/switchtec.c

```diff
@@ -1333,36 +1333,64 @@ static int log_c_to_file(struct switchtec_dev *dev, int sub_cmd_id, int fd)
 	return 0;
 }
 
+static int log_ram_flash_to_file(struct switchtec_dev *dev,
+				 int subcmd1, int subcmd2, int subcmd3,
+				 int fd, FILE *log_def_file,
+				 struct switchtec_log_file_info *info)
+{
+	int ret;
+	int subcmd;
+
+	if (switchtec_is_gen5(dev)) {
+		subcmd = subcmd1;
+		return log_a_to_file(dev, subcmd, fd, log_def_file,
+				     info);
+	}
+	else {
+		subcmd = subcmd2;
+		ret = log_a_to_file(dev, subcmd, fd, log_def_file,
+				    info);
+		if (ret > 0) {
```

Comment:
Shouldn't  this be < 0? Or rather we should be checking for the error specifically saying that the subcmd was not accepeted.

Though, the error handling in log_a_to_file seems to be a mess..... Would be better to follow the convention of returning -1 on error and setting errno.

Reply:
Yes, in fact we should check for all errors, so it's updated to `if(ret)` now.